include/scan.hpp

```diff
@@ -2,13 +2,57 @@
 
 #include "parse.hpp"
 #include "types.hpp"
+#include <format>
+#include <tuple>
 
 namespace stdx {
 
-// замените болванку функции scan на рабочую версию
 template <typename... Ts>
 std::expected<details::scan_result<Ts...>, details::scan_error> scan(std::string_view input, std::string_view format) {
-    return std::unexpected(details::scan_error{"Dumb implementation"});
+    const auto string_values = details::parse_sources<Ts...>(input, format);
+
+    if (!string_values.has_value()) {
+        return std::unexpected(std::move(string_values.error()));
+    }
+
+    const auto &formats = string_values->first;
+    const auto &inputs = string_values->second;
+
```

Comment:
Чуть более короткая запись
```suggestion
    const auto& [formats, inputs] = string_values.value();

```